### DISCUSSION TITLE
Thoughts on forking the Crystal parser?

### DISCUSSION THREAD

**Discussion by FnControlOption** [veelenga/ameba#308] [Ideas]
Understandably, it takes awhile for some PRs to be merged in crystal-lang/crystal such as https://github.com/crystal-lang/crystal/pull/11798. What do you guys think about vendoring a fork of the parser? Here's a repo/shard where I've separated the parser code from the rest of the standard library: https://github.com/FnControlOption/crystal-syntax

**Comment by Sija** [comment]
In general I don't think that forking is a good idea. I'd use this method as the last resort. This approach takes a lot of energy, is prone to bugs and requires a lot of maintenance. I'd rather put more energy into quickly merging these PRs instead :)

**Comment by straight-shoota** [comment]
Maybe a good alternative would be to vendor in the compiler source and pin it to a specific version. Thus ameba would use the parser code from the compiler (no need to maintain another one), but with its own copy which is not necessarily always the latest version (no need for hotfixes to keep up with breaking changes upstream).
